### PR TITLE
Add hanota in C code

### PR DESCRIPTION
**Pull request by Zuoxun**
Running results are as follows:

> 初始状态下：
A =[5,4,3,2,1]
B =
C =
圆盘移动完成后：
A =
B =
C =[5,4,3,2,1]

If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by Zuoxun** [comment]
Code hanota.c would rely on utils/vector.h, so this PR has already included the content of PR “Update vector.h” .

File: codes/c/chapter_divide_and_conquer/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+add_executable(hanota.c)
```
Comment (Gonglja) [inline codes/c/chapter_divide_and_conquer/CMakeLists.txt:1]: The `add_executable` syntax is 
```cmake
add_executable(<name> [source1] [source2 ...])
```
Reference: https://cmake.org/cmake/help/latest/command/add_executable.html

File: codes/c/chapter_divide_and_conquer/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+add_executable(binary_search_recur.c)
```
Comment (Gonglja) [inline codes/c/chapter_divide_and_conquer/CMakeLists.txt:2]: Temporarily removed, this PR has nothing to do with `binary_search_recur`

File: codes/c/utils/vector.h
Comment (Gonglja) [inline codes/c/utils/vector.h:1]: Repeat with #818

File: codes/c/chapter_divide_and_conquer/hanota.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * Created Time: 2023-10-01
+ * Author: Zuoxun (845242523@qq.com)
+ */
+
+#include "../utils/common.h"
+
+/* 移动一个圆盘 */
+void move(vector *src, vector *tar) {
+    // 从 src 顶部拿出一个圆盘
+    int *pan_temp = vectorBack(src);
```
Comment (Gonglja) [inline codes/c/chapter_divide_and_conquer/hanota.c:12]: `panTemp`

File: codes/c/chapter_divide_and_conquer/hanota.c
Comment (Gonglja) [inline codes/c/chapter_divide_and_conquer/hanota.c:1]: Please add running results

**Comment by Gonglja** [comment]
Can be merged @krahets

**Review by krahets — APPROVED** [review]
Thanks @Zuoxun  @Gonglja !

**Pull request merged**